src/commands.rs

```diff
@@ -455,28 +431,20 @@ fn print_stats(stats: CacheStats) -> Result<()> {
 
 /// Send a `Compile` request to the server, and return the server response if successful.
 fn request_compile<W: AsRef<Path>, X: AsRef<OsStr>, Y: AsRef<Path>>(conn: &mut ServerConnection, exe: W, args: &Vec<X>, cwd: Y) -> Result<CompileResponse> {
-    //TODO: It'd be nicer to send these over as raw bytes.
-    let exe = exe.as_ref().to_str().ok_or("bad exe")?;
-    let cwd = cwd.as_ref().to_str().ok_or("bad cwd")?;
-    let args = args.iter().filter_map(|a| a.as_ref().to_str().map(|s| s.to_owned())).collect::<Vec<_>>();
-    if args.is_empty() {
-        bail!("bad commandline")
-    }
-    let mut req = ClientRequest::new();
-    let mut compile = Compile::new();
-    compile.set_exe(exe.to_owned());
-    compile.set_cwd(cwd.to_owned());
-    compile.set_command(RepeatedField::from_vec(args));
-    trace!("request_compile: {:?}", compile);
-    req.set_compile(compile);
+    let req = Request::Compile(Compile {
+        exe: exe.as_ref().to_str().ok_or("bad exe")?.to_owned(),
```

Comment:
as a follow-up, we could now just use `OsStr` or `OsString` all the way for the values from the client commandline.